closure/private/files_equal_test.bzl

```diff
@@ -0,0 +1,61 @@
+# -*- mode: python; -*-
+#
+# Copyright 2016 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests that two files contain the same binary data."""
+
+def _impl(ctx):
+  ctx.file_action(
+      output=ctx.outputs.executable,
+      content="\n".join([
+          "#!/bin/bash",
+          "function checksum() {",
+          "  if command -v sha256sum >/dev/null; then",
+          "    sha256sum $1 | cut -f 1 -d ' '",
+          "  elif command -v shasum >/dev/null; then",
+          "    cat $1 | shasum -a 256 | cut -f 1 -d ' '",
+          "  else",
+          "    echo no sha256 checksum command found >&2",
+          "    exit 1",
+          "  fi",
+          "}",
+          "SUM1=$(checksum %s)" % ctx.file.golden.short_path,
+          "SUM2=$(checksum %s)" % ctx.file.actual.short_path,
+          "if [[ ${SUM1} != ${SUM2} ]]; then",
+          "  echo FILES DO NOT HAVE EQUAL CONTENTS >&2",
+          "  echo %s ${SUM1} >&2" % ctx.file.golden.short_path,
+          "  echo %s ${SUM2} >&2" % ctx.file.actual.short_path,
+          "  exit 1",
+          "fi",
+      ]),
```

Comment:
Are you sure you're not mistaken? I'm passing the format arguments in via `%`. I also can't really use `"\n".join([...]).format(golden=..., ...)` here because then it will complain that all the other stuff inside curlies doesn't have a replacement defined.